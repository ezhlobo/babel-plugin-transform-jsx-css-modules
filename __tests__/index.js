const pluginTester = require('babel-plugin-tester')
const plugin = require('../index.js')

pluginTester({
  plugin,
  title: 'default config',
  tests: [
    {
      title: 'does not change files without styles',
      code: '<div className="global" styleName="local"></div>;',
    },

    {
      title: 'amends class names if styles import is provided',
      snapshot: true,
      code: `
        import './styles.css';

        <div test="test" styleName="local"></div>;
        <div test="test" styleName="local-one local-two"></div>;
      `,
    },

    {
      title: 'does not change import if styles is imported already',
      snapshot: true,
      code: `
        import __cssmodule__ from './styles.css';

        <div styleName="local"></div>;
        <div styleName="local-one local-two"></div>;
      `,
    },

    {
      title: 'combines global and local styles',
      snapshot: true,
      code: `
        import __cssmodule__ from './styles.css';

        <div className="global" styleName="local"></div>;
      `,
    }
  ],
})

pluginTester({
  plugin,
  pluginOptions: {
    sourceAttribute: 'className',
  },
  title: 'with options: {sourceAttribute: "className"}',
  tests: [
    {
      title: 'combines global and local styles',
      snapshot: true,
      code: `
        import __cssmodule__ from './styles.css';

        <div className="global"></div>;
      `,
    }
  ],
})
