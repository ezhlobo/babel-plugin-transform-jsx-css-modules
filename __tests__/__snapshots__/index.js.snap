// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default config amends class names if styles import is provided: amends class names if styles import is provided 1`] = `
"
import './styles.css';

<div test=\\"test\\" styleName=\\"local\\"></div>;
<div test=\\"test\\" styleName=\\"local-one local-two\\"></div>;

      ↓ ↓ ↓ ↓ ↓ ↓

import __cssmodule__ from \\"./styles.css\\";


<div test=\\"test\\" className={__cssmodule__[\\"local\\"]}></div>;
<div test=\\"test\\" className={[__cssmodule__[\\"local-one\\"], __cssmodule__[\\"local-two\\"]]}></div>;
"
`;

exports[`default config combines global and local styles: combines global and local styles 1`] = `
"
import __cssmodule__ from './styles.css';

<div className=\\"global\\" styleName=\\"local\\"></div>;

      ↓ ↓ ↓ ↓ ↓ ↓

import __cssmodule__ from './styles.css';

<div className={[\\"global\\", __cssmodule__[\\"local\\"]]}></div>;
"
`;

exports[`default config does not change import if styles is imported already: does not change import if styles is imported already 1`] = `
"
import __cssmodule__ from './styles.css';

<div styleName=\\"local\\"></div>;
<div styleName=\\"local-one local-two\\"></div>;

      ↓ ↓ ↓ ↓ ↓ ↓

import __cssmodule__ from './styles.css';

<div className={__cssmodule__[\\"local\\"]}></div>;
<div className={[__cssmodule__[\\"local-one\\"], __cssmodule__[\\"local-two\\"]]}></div>;
"
`;

exports[`with options: {sourceAttribute: "className"} combines global and local styles: combines global and local styles 1`] = `
"
import __cssmodule__ from './styles.css';

<div className=\\"global\\"></div>;

      ↓ ↓ ↓ ↓ ↓ ↓

import __cssmodule__ from './styles.css';

<div className={__cssmodule__[\\"global\\"]}></div>;
"
`;
